GitExtUtils/Linq/LinqExtensions.cs

```diff
@@ -75,9 +76,10 @@ public static class LinqExtensions
             foreach (TSource sourceElement in source)
             {
                 TKey key = keySelector(sourceElement);
+
                 if (key == null)
                 {
-                    var ex = new ArgumentNullException("KeySelector produced a key that is null. See exception data for source.");
+                    var ex = new Exception("KeySelector produced a key that is null. See exception data for source.");
```

Comment:
Two usages here are not validating arguments. They're validating the result of `keySelector`. Opted for `Exception` as that's what a lot of existing code does. Could be `InvalidOperationException`, but I'm indifferent.

Reply:
Specifically, the first argument to `ArgumentNullException` is supposed to be a parameter name. Given there's no parameter here, it's not possible to use that exception type here.

Reply:
:bulb: You could use `ArgumentException` here since `keySelector` does not behavior in a manner consistent with the requirements of this method.